server/src/main/java/org/apache/cassandra/sidecar/cdc/CdcManager.java

```diff
@@ -0,0 +1,282 @@
+                                                                      sidecarInstancesProvider,
+                                                                      secretsProvider,
+                                                                      clientConfig,
+                                                                      conf,
+                                                                      cdcStats,
+                                                                      taskExecutorPool);
+                                    }
+                                    catch (IOException e)
+                                    {
+                                        throw new RuntimeException(e);
+                                    }
+                                });
+                            }))
+                   .collect(Collectors.toList());
+
+        consumers = new ArrayList<>(uniqueConsumers.values());
+        return consumers;
+    }
+
+    SidecarCdc loadOrBuildCdcConsumer(Integer instanceId,
+                                      ClusterConfigProvider clusterConfigProvider,
+                                      EventConsumer eventConsumer,
+                                      SchemaSupplier schemaSupplier,
+                                      TokenRangeSupplier tokenRangeSupplier,
+                                      CdcSidecarInstancesProvider sidecarInstancesProvider,
+                                      SecretsProvider secretsProvider,
+                                      SidecarCdcClient.ClientConfig clientConfig,
+                                      CdcConfig conf,
+                                      ICdcStats cdcStats,
+                                      TaskExecutorPool taskExecutorPool) throws IOException
+    {
+        return buildConsumer(conf.jobId(),
+                             instanceId,
+                             new SidecarCdcOptions(instanceFetcher),
+                             clusterConfigProvider,
+                             eventConsumer,
+                             schemaSupplier,
+                             tokenRangeSupplier,
+                             sidecarInstancesProvider,
+                             clientConfig,
+                             secretsProvider,
+                             cdcStats,
+                             taskExecutorPool);
+    }
+
+    public void startConsumers()
+    {
+        consumers.forEach(SidecarCdc::initSchema);
+        consumers.forEach(SidecarCdc::start);
+    }
+
+    public void stopConsumers()
+    {
+        consumers.forEach(SidecarCdc::stop);
+    }
+
+    private Integer getInstanceId(String instanceIp)
+    {
+        for (InstanceMetadata instance : instanceFetcher.allLocalInstances())
+        {
+            String configuredHost = instance.ipAddress();
```

Comment:
This is an ip address, no?

```suggestion
            String configuredIpAddress = instance.ipAddress();
```